src/ensembl/src/content/app/browser/browser-region-field/BrowserRegionField.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState, FormEvent, useEffect } from 'react';
+import { replace, Replace } from 'connected-react-router';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+import Input from 'src/shared/components/input/Input';
+import Tooltip from 'src/shared/components/tooltip/Tooltip';
+
+import { ChrLocation, RegionValidationResponse } from '../browserState';
+import { RootState } from 'src/store';
+import {
+  changeBrowserLocation,
+  resetRegionValidation,
+  toggleRegionFieldActive,
+  validateRegion
+} from '../browserActions';
+import {
+  getBrowserActiveGenomeId,
+  getRegionEditorActive,
+  getRegionFieldActive,
+  getRegionValidationInfo,
+  getRegionValidationLoadingStatus,
+  getChrLocation
+} from '../browserSelectors';
+import { getIsDrawerOpened } from '../drawer/drawerSelectors';
+import {
+  getChrLocationFromStr,
+  getRegionFieldErrorMessages
+} from '../browserHelper';
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import { LoadingState } from 'src/shared/types/loading-state';
+
+import applyIcon from 'static/img/shared/apply.svg';
+import clearIcon from 'static/img/shared/clear.svg';
+
+import styles from './BrowserRegionField.scss';
+import browserStyles from '../Browser.scss';
+import browserNavBarStyles from '../browser-nav/BrowserNavBar.scss';
+
+type BrowserRegionFieldProps = {
+  activeGenomeId: string | null;
+  chrLocation: ChrLocation;
+  isDrawerOpened: boolean;
+  regionEditorActive: boolean;
+  regionFieldActive: boolean;
```

Comment:
Let's go over the props here.
Why does BrowserRegionField need to know stuff about other components (that the Drawer is opened or that the regionEditor is active)? Whether the Drawer is open or not, should be of no concern for the BrowserRegionField. What it should care about is whether it's active or not, and whether it is disabled (covered with overlay )or not. Which, I think, suggests that the props should be `isActive` and `isDisabled`.

Same goes for BrowserRegionEditor — its props also are informing it of the state of other components, which should be none of its business.

Reply:
The BrowserRegionField has to know whether the BrowserRegionEditor is opened as it should have an overlay on top of it to make it not clickable, when the latter is active. So if we are to store the status instead for BrowserRegionField, we will need to toggle the status of it in the Redux action that sets the active status for BrowserRegionEditor. Will this be a better approach? Same applies to the drawer status.